lib/structured_data.rb

```diff
@@ -0,0 +1,73 @@
+require 'nokogiri'
+require 'erb'
+
+# This creates a Structured Data FAQ section for google search results
+# https://developers.google.com/search/docs/data-types/faqpage
+#
+# The page needs to be annotated with the attribute `structured_data`
+# and questions and answers formatted with a `.section-faq`.
+# See `/articles/faq-ssl-certificates/` for an example.
+class AddStructuredDataFilter < Nanoc::Filter
+
+  identifier :add_structured_data
+
+  def run(content, params={})
+    if @item[:structured_data]
+      doc = Nokogiri::HTML(content)
+      write_faqs(doc, collect_faqs(doc))
+      return doc.to_html
+    end
+    content
+  end
+
+  def write_faqs(doc, faqs)
+    head = doc.at('head')
```

Comment:
You can move this line after the template generation (as there is no need to have it here and makes the logic easier to follow.

Reply:
✅